### PR TITLE
Avoid false positive for missing return

### PR DESCRIPTION
**Pull request by xgouchet**
Given a function with `throw` statements in some branches, there is a false positive that considers those branches as invalid. A `throw` can be a valid exit strategy for a function when "something truly terrible happens". 

```
function ok7() as integer
    a = 1
    if a > 0
        throw "something wrong"
    else
        return 0
    end if
end function
```

### CODE REVIEW

**Comment by xgouchet** [comment]
This also upgrades the Brighterscript dependency to `0.59.0` to use the `isThrowStatement` method.

**Comment by xgouchet** [comment]
Seems like some tests are failing in CI with the 0.59.0 upgrade, I'll fix those asap 👍

File: test/project1/source/consistent-return.brs [hunk 1]
```diff
@@ -46,6 +46,15 @@ sub error9()
     end sub)
 end sub
 
+function error10() as integer
+    a = 1
+    if a > 0
+        throw "something wrong"
+    else
+        return 0' missing return or throw here
```
Comment (elsassph) [inline test/project1/source/consistent-return.brs:54]: Why would that case be an error? It's the same code as `ok7`, no?

**Comment by elsassph** [comment]
Thanks, the `return` tracking was implemented before `throw` was introduced!

File: test/project1/source/consistent-return.brs (hunk 1 above)
Comment (xgouchet) [reply test/project1/source/consistent-return.brs:54]: My bad the `return` shouldn't be there you're right

Comment (elsassph) [reply test/project1/source/consistent-return.brs]: `throw` should be also valid for not returning anything.

Comment (xgouchet) [reply test/project1/source/consistent-return.brs]: Yes but it's not a void  function here.

**Comment by elsassph** [comment]
Can you check if it fixes #68

**Comment by xgouchet** [comment]
> Can you check if it fixes #68

No it doesn't but I'll add that to my PR

File: test/project1/source/consistent-return.brs (hunk 1 above)
Comment (elsassph) [reply test/project1/source/consistent-return.brs]: I know but is this OK too?
```
function ok() as void
    a = 1
    if a > 0
        throw "something wrong"
    end if
end function
```

Comment (xgouchet) [reply test/project1/source/consistent-return.brs]: Oh sorry, yes this one works fine, I'll add it in the test cases

**Comment by elsassph** [comment]
Code looks good, but needs testing at scale.

**Comment by TwitchBronBron** [comment]
I can test this on our app today.

**Comment by TwitchBronBron** [comment]
Sorry for the delay. I finally got around to testing this, and it looks like it works well against our larger project. the only change I would suggest is bumping the `peerDependency` value up since this will fail for older brighterscript versions.

**Pull request merged**